src/governance.cpp

```diff
@@ -211,6 +211,29 @@ void CGovernanceManager::ProcessMessage(CNode* pfrom, std::string& strCommand, C
 
         if(!fIsValid) {
             if(fMasternodeMissing) {
+
```

Comment:
:+1: 

This is a very interesting idea. By checking the MN collateral in the blockchain directly you can completely bypass the masternode list, which is inconsistent.  This would fix some problems we have observed with objects being rejected because the MN could not be found.

However this isn't the place to do it. This check should be added in governance-object.cpp, replacing the current MN check.